win32ss/user/user32/misc/exticon.c

```diff
@@ -714,9 +714,75 @@ static UINT ICO_ExtractIconExW(
 	    xresdir = find_entry_by_id(iconresdir, LOWORD(pIconId[i]), rootresdir);
             if( !xresdir )
             {
+#ifdef __REACTOS__
+              const IMAGE_RESOURCE_DIRECTORY *resdir;
+              WARN("icon entry %d not found\n", LOWORD(pIconId[i]));
+
+              /* Try to get an icon by walking the files Resource Section */
+              if (iconresdir->NumberOfIdEntries > 0)
+              {
+                  xresent = (const IMAGE_RESOURCE_DIRECTORY_ENTRY*)
+                      (ULONG_PTR)(iconresdir + 1);
+              }
+              else
+              {
+                  RetPtr[i]=0;
+                  continue;
+              }
+
+              /* Get the Resource Directory */
+              resdir = (const IMAGE_RESOURCE_DIRECTORY *)
+                       ((const char *)rootresdir + xresent->OffsetToDirectory);
+
+              if (resdir->NumberOfIdEntries > 0) // Do we have entries
+              {
+                  xresent = (const IMAGE_RESOURCE_DIRECTORY_ENTRY*)
+                      (ULONG_PTR)(resdir + 1);
+              }
+              else
+              {
+                  RetPtr[i]=0;
+                  continue;
+              }
+
+              /* Retreive the data entry and find its address */
+              igdataent = (const IMAGE_RESOURCE_DATA_ENTRY*)
+                  ((const char *)rootresdir + xresent->OffsetToData);
+              idata = RtlImageRvaToVa(RtlImageNtHeader((HMODULE)peimage),
+                  (HMODULE)peimage, igdataent->OffsetToData, NULL);
+              if (!idata)
+              {
+                  RetPtr[i]=0;
+                  continue;
+              }
+
+              /* Check to see if this looks like an icon bitmap */
+              if (idata[0] == sizeof(BITMAPINFOHEADER))
+              {
+                  BITMAPINFOHEADER bmih;
+                  RtlCopyMemory(&bmih, idata, sizeof(BITMAPINFOHEADER));
+                  /* Do the Width and Height look correct for an icon */
+                  if (bmih.biWidth == 32 && bmih.biHeight == 64)
```

Comment:
Is a valid icon 32 x 64 only ?
Is it really a && ? Or || using both dimensions in check ?

Reply:
This checks for a valid 32x32 icon only. The height is twice as much because there is one full sized map for the AND icon (actual image) and another full sized map for the XOR (transparency map). We want to check for both of these sizes being correct before we accept the icon as valid. Thanks.

Reply:
```suggestion
                  if ((bmih.biWidth * 2) == bmih.biHeight)
```

Reply:
Good Idea. Done. Thanks.